Restore onConfirm/onCancel props for programmatically opening a dialog

kikuomax opened the pull request:
@ElteHupkes and @kyle-jennings, thanks for your PRs:
- #1
- #4

As @ElteHupkes [commented](https://github.com/kikuomax/buefy/pull/4#issuecomment-1403650776), I intentionally replaced `onConfirm` and `onCancel` props of `Dialog` with `confirmCallback` and `cancelCallback`, because they conflict with event listeners for `@confirm` and `@cancel` on Vue 3.

However, I came up with a workaround to keep my fork better compatible with the original buefy. With this PR merged, you can specify `onConfirm` and `onCancel` to props for programmatically opening a `Dialog` or `Modal` as you do on the original buefy.

kikuomax committed:
fix(lib): restore onConfirm and onCancel

- Restores the `onConfirm` and `onCancel` properties given to
  programmatic `Dialog.open` because I came up with a better workaround.
  `Dialog` continues to accept `confirmCallback` and `cancelCallback`
  instead of `onConfirm` and `onCancel` but `Dialog.open` forwards
  `confirmCallback` and `cancelCallback` calls to `propsData.onConfirm`
  and `propsData.onCancel` respectively.
  Restores also the `onCancel` property given to programmatic
  `Modal.open`. `Modal` still needs `cancelCallback` instead of
  `onCancel`.

issue kikuomax/buefy#4
Co-authored-by: kylejennings83 <kyle.jennings@traliant.com>

kikuomax committed:
fix(docs): restore onConfirm of Dialog.confirm/prompt

- Replaces `confirmCallback` given to `dialog.confirm`, and
  `dialog.prompt` with `onConfirm` because the `onConfirm` property was
  restored in the last commit.

kikuomax committed:
docs: update CHANGELOG and MIGRATION-NOTE

- `CHANGELOG.md` and `MIGRAION-NOTE.md` explain the restoration of
  the `onConfirm` and `onCancel` properties given to the function to
  programmatically open a `Dialog`, and the `onCancel` property to the
  function to programmatically open a `Modal`.

kyle-jennings commented:
@kikuomax do you need us to test this out in order to merge it in?

benrugg commented:
This is a good call! I just wrestled with this for a couple hours, thinking there must be a bug in your port of Buefy, until I finally managed to come across this pull request! 

If you don't/can't pull this in soon, I might suggest adding a bold section on your readme to show the handful of differences between the original Buefy and this version. (I just now saw that this is in the changelog, but it feels pretty buried!)

Separately, THANK YOU for making this! It's so helpful and good for the community!

kikuomax committed:
docs: highlight the change → v0.9.21-vue3-2

- `README.md` highlights the change from the tag `v0.9.21-vue3-1` to
  `v0.9.21-vue3-2`.

kikuomax commented:
@kyle-jennings Thanks! I appreciate your generous offer, but I am going to merge this PR anyway.
